dateutil/parser/_parser.py

```diff
@@ -1153,9 +1146,18 @@ def _build_tzaware(self, naive, res, tzinfos):
         elif res.tzoffset:
             aware = naive.replace(tzinfo=tz.tzoffset(res.tzname, res.tzoffset))
 
-        else:
-            # TODO: this is really only the right thing to do if no tz
-            # information was found.
+        elif not res.tzname and not res.tzoffset:
+            # i.e. no timezone information was found.
+            aware = naive
+
+        elif res.tzname:
+            # tz-like string was parsed but we don't know what to do
+            # with it
+            warnings.warn("tzname {tzname} identified but not understood.  "
```

Comment:
We should raise a specific warning type so that it can be easily ignored by people who don't want to see this message. I suggest subclassing maybe `RuntimeWarning`, maybe `UnknownTimezoneWarning(RuntimeWarning)`?

Reply:
Also, I wouldn't specify `ValueError` in the message. Just say it will raise an error.

Reply:
> Also, I wouldn't specify `ValueError` in the message. Just say it will raise an error.

But the error (in the future) will be a ValueError (or a subclass), so the users seeing it can already prepare for it.

Reply:
@wagner-certat It might be a `ValueError`, but probably it will be a subclass. I'm not quite sure how they would "prepare for it" using just the information that it might raise a `ValueError`, though, since that would catch nearly *all* the errors raised by `parse`.

Reply:
True, I also catch all `ValueError`s already now.